x-pack/plugins/fleet/server/services/package_policies/experimental_datastream_features.ts

```diff
@@ -171,12 +184,24 @@ export async function handleExperimentalDatastreamFeatureOptIn({
         },
       };
 
+      updatedIndexTemplate = indexTemplateBody as IndexTemplate;
+
       await esClient.indices.putIndexTemplate({
         name: featureMapEntry.data_stream,
         // @ts-expect-error
         body: indexTemplateBody,
       });
     }
+
+    updatedIndexTemplates.push({
+      templateName: featureMapEntry.data_stream,
+      indexTemplate: updatedIndexTemplate,
+    });
+  }
+
+  // Trigger rollover for updated datastreams
+  if (updatedIndexTemplates.length > 0) {
+    await updateCurrentWriteIndices(esClient, appContextService.getLogger(), updatedIndexTemplates);
```

Comment:
I wonder why we pass `logger` here as parameter, usually `appContextService.getLogger()` is imported where it is needed.

Reply:
I think we have both pattern in Fleet code some service need logger as a dependency (mostly EPM methods) and some use appContext, I only consume the method here so it's probably not something that need to be changed in that PR.